lib/Runtime/Library/JavascriptError.cpp

```diff
@@ -53,19 +53,16 @@ namespace Js
         SetEnumerable(propertyId, false);
     }
 
-    Var JavascriptError::NewInstance(RecyclableObject* function, JavascriptError* pError, CallInfo callInfo, Arguments args)
+    Var JavascriptError::NewInstance(RecyclableObject* function, JavascriptError* pError, CallInfo callInfo, Var newTarget, Var message)
     {
         ScriptContext* scriptContext = function->GetScriptContext();
 
-        AssertMsg(args.Info.Count > 0, "Should always have implicit 'this'");
-
-        Var newTarget = callInfo.Flags & CallFlags_NewTarget ? args.Values[args.Info.Count] : args[0];
-        bool isCtorSuperCall = (callInfo.Flags & CallFlags_New) && (callInfo.Flags & (CallFlags_Value|CallFlags_NewTarget)) && newTarget != nullptr && RecyclableObject::Is(newTarget);
+        bool isCtorSuperCall = (callInfo.Flags & (CallFlags_New|CallFlags_NewTarget)) == (CallFlags_New|CallFlags_NewTarget) && newTarget != nullptr && !JavascriptOperators::IsUndefined(newTarget);
```

Comment:
Good point. @boingoing we discussed the fact that this should be checking for undefined. Should we change the check in the other built-ins? RecyclableObject::Is will return false for tagged numbers but true for undefined.